packages/flutter_tools/lib/src/ios/xcodeproj.dart

```diff
@@ -477,16 +480,29 @@ class XcodeProjectInfo {
     }
     return false;
   }
+
   /// Returns unique scheme matching [buildInfo], or null, if there is no unique
   /// best match.
-  String? schemeFor(BuildInfo? buildInfo) {
-    final String expectedScheme = expectedSchemeFor(buildInfo);
-    if (schemes.contains(expectedScheme)) {
-      return expectedScheme;
+  String? schemeFor({String? flavor, String hostAppProjectName = ''}) {
```

Comment:
i actually think it should be mandatory? because this function needs to know what the project is called (either Runner or a custom name) and anything calling this function wouldn't be aware if the project was renamed.